### DISCUSSION TITLE
why start clickhouse server reports error in termux debian

### DISCUSSION THREAD

**Discussion by l1t1** [ClickHouse/ClickHouse#49082] [Q&A] [answered]
root@localhost:~# /data/data/com.termux/files/home/ck server &
[1] 29344
root@localhost:~# Processing configuration file 'config.xml'.
There is no file 'config.xml', will use embedded config.
2023.04.23 22:11:26.324151 [ 29344 ] {} <Trace> Pipe: Pipe capacity is 1.00 MiB
2023.04.23 22:11:26.717863 [ 29344 ] {} <Information> Application: Starting ClickHouse 23.4.1.1073 (revision: 54473, git hash: 9eee7c72fe63ffee07b8bb18359633876d5effa1, build id: 2D82F9B9F260453A4573EED738CD0DB9B342A6C6), PID 29344
2023.04.23 22:11:26.719357 [ 29344 ] {} <Information> Application: starting up
2023.04.23 22:11:26.719686 [ 29344 ] {} <Information> Application: OS name: Linux, version: 6.2.1-PRoot-Distro, architecture: aarch64
2023.04.23 22:11:26.758734 [ 29344 ] {} <Information> Application: Available RAM: 7.56 GiB; physical cores: 8; logical cores: 8.
2023.04.23 22:11:26.898959 [ 29344 ] {} <Trace> AsynchronousMetrics: Scanning /sys/class/thermal
2023.04.23 22:11:26.899183 [ 29344 ] {} <Trace> AsynchronousMetrics: Scanning /sys/block
2023.04.23 22:11:26.924595 [ 29344 ] {} <Error> Application: std::exception. Code: 1001, type: std::__1::__fs::filesystem::filesystem_error, e.what() = filesystem error: in directory_iterator::directory_iterator(...): Permission denied ["/sys/block"], Stack trace (when copying this message, always include the lines below):

0. std::runtime_error::runtime_error(String const&) @ 0x1824d3d8 in /data/data/com.termux/files/home/ck
1. std::system_error::system_error(std::error_code, String const&) @ 0x182563c0 in /data/data/com.termux/files/home/ck
2. ? @ 0xff49930 in /data/data/com.termux/files/home/ck
3. ? @ 0x181f4678 in /data/data/com.termux/files/home/ck
4. ? @ 0x181f2e60 in /data/data/com.termux/files/home/ck
5. std::__fs::filesystem::directory_iterator::directory_iterator(std::__fs::filesystem::path const&, std::error_code*, std::__fs::filesystem::directory_options) @ 0x181f2d38 in /data/data/com.termux/files/home/ck
6. DB::AsynchronousMetrics::openBlockDevices() @ 0xffd998c in /data/data/com.termux/files/home/ck
7. DB::AsynchronousMetrics::AsynchronousMetrics(int, std::function<std::vector<DB::ProtocolServerMetrics, std::allocator<DB::ProtocolServerMetrics>> ()> const&) @ 0xffd8c18 in /data/data/com.termux/files/home/ck
8. DB::ServerAsynchronousMetrics::ServerAsynchronousMetrics(std::shared_ptr<DB::Context const>, int, int, std::function<std::vector<DB::ProtocolServerMetrics, std::allocator<DB::ProtocolServerMetrics>> ()> const&) @ 0x14aaad34 in /data/data/com.termux/files/home/ck
9. DB::Server::main(std::vector<String, std::allocator<String>> const&) @ 0xff80fa4 in /data/data/com.termux/files/home/ck
10. Poco::Util::Application::run() @ 0x1653de9c in /data/data/com.termux/files/home/ck
11. DB::Server::run() @ 0xff7e190 in /data/data/com.termux/files/home/ck
12. Poco::Util::ServerApplication::run(int, char**) @ 0x16552ebc in /data/data/com.termux/files/home/ck
13. mainEntryClickHouseServer(int, char**) @ 0xff7be54 in /data/data/com.termux/files/home/ck
14. main @ 0xbb7bd80 in /data/data/com.termux/files/home/ck
15. ? @ 0x273fc in /lib/aarch64-linux-gnu/libc.so.6
16. __libc_start_main @ 0x274cc in /lib/aarch64-linux-gnu/libc.so.6

Cannot print extra info for Poco::Exception (version 23.4.1.1073 (official build))
2023.04.23 22:11:26.925039 [ 29344 ] {} <Error> Application: filesystem error: in directory_iterator::directory_iterator(...): Permission denied ["/sys/block"]
2023.04.23 22:11:26.925066 [ 29344 ] {} <Information> Application: shutting down
2023.04.23 22:11:26.925082 [ 29344 ] {} <Debug> Application: Uninitializing subsystem: Logging Subsystem
2023.04.23 22:11:26.925466 [ 29345 ] {} <Trace> BaseDaemon: Received signal -2
2023.04.23 22:11:26.925512 [ 29345 ] {} <Information> BaseDaemon: Stop SignalListener thread

[1]+  Exit 70                 /data/data/com.termux/files/home/ck server

**Comment by l1t1** [comment]
clickhouse local works
root@localhost:~# /data/data/com.termux/files/home/ck local
ClickHouse local version 23.4.1.1073 (official build).

localhost.localdomain :) show databases;

SHOW DATABASES

Query id: dac822b8-1285-4089-91b3-03ea76f5ac52

┌─name───────────────┐
│ INFORMATION_SCHEMA │
│ _local             │
│ information_schema │
│ system             │
└────────────────────┘

4 rows in set. Elapsed: 0.005 sec.

localhost.localdomain :)

**Comment by l1t1** [comment]
in another machine, also cannot start but in differenct position.
![clickhousetermux2](https://user-images.githubusercontent.com/23301704/233874803-a3dee44d-24e3-4564-a096-cfeb8edeb310.jpg)

**Comment by mlex** [comment, marked as answer]
ClickHouse server polls `/sys/block` and `/sys/class/hwmon` (and other system files) regularly to fill the `system.asynchronous_metrics` and `system.asynchronous_metric_log` tables. There is no way to disable that polling (yet). Seems like on temux those files exist, but are not readable. Which leads to the error on startup.

`clickhouse-local` doesn't collect data for those system tables. That's why it starts fine.

**Comment by l1t1** [comment]
thanks

**Comment by l1t1** [comment]
wish add a way to disable those checks
